dataikuapi/dss/ml.py

```diff
@@ -716,47 +798,39 @@ def __repr__(self):
 
-    def set_values(self, values=None):
-        if values is None:
-            warnings.warn("Categorical hyperparameter \"{}\" not modified".format(self.name))
-        else:
-            assert isinstance(values, dict), "Invalid values input type for hyperparameter " \
-                                             "\"{}\": ".format(self.name) + \
-                                             "must be a dictionary"
-            admissible_values = self._algo_settings[self.name]["values"].keys()
-            for category, setting in values.items():
-                assert category in admissible_values, "Unknown categorical value \"" + category + "\". Expected a member of " + str(list(admissible_values))
-                value_error_message = "Invalid input value for hyperparameter \"{}\", category \"{}\": ".format(self.name, category)
-                value_error_message += "expected a {\"enabled\": bool} dictionary"
-                assert isinstance(setting, dict), value_error_message
-                assert list(setting.keys()) == ["enabled"], value_error_message
-                assert all(type(v) == bool for v in setting.values()), value_error_message
-            for category, setting in values.items():
-                self._algo_settings[self.name]["values"][category] = setting
-
-    def enable_categories(self, categories, disable_others=False):
-        accepted_categories = self._algo_settings[self.name]["values"].keys()
-        for category in categories:
-            assert isinstance(category, string_types)
-            assert category in accepted_categories
-        self.set_values({category: {"enabled": True}
-                         for category in categories})
-        if disable_others:
-            self.set_values({category: {"enabled": False}
-                             for category in accepted_categories
-                             if category not in categories})
-
-    def disable_categories(self, categories, enable_others=False):
-        accepted_categories = self._algo_settings[self.name]["values"].keys()
-        for category in categories:
-            assert isinstance(category, string_types)
-            assert category in accepted_categories
-        self.set_values({category: {"enabled": False}
-                         for category in categories})
-        if enable_others:
-            self.set_values({category: {"enabled": True}
-                             for category in accepted_categories
-                             if category not in categories})
+    def set_values(self, values):
+        """
+        Enables the search over listed values (categories).
+        :param values: values to enable, all other values will be disabled
+        :type values: list of str
+        """
+        all_possible_values = self.get_all_possible_values()
+        for category in values:
```

Comment:
Add a check that values is indeed a `list`. If user wants to pass a single value (like me when testing 😬), the displayed error is complex to understand, as we iterate over the characters in that case.

```suggestion
        all_possible_values = self.get_all_possible_values()
        assert isinstance(values, list), \
            "Invalid input type {} for categorical hyperparameter {}: must be a list of strings".format(type(values), self.name)
        for category in values:
```

Reply:
Good point, but we probably want to allow other iterables ? IIRC I had removed this check for that reason here (or maybe somewhere else).

Reply:
To make things easy, we can just check that values is not some kind of string to allow people pass fancy iterables.